packages/hub/routes/scheduled-payments.ts

```diff
@@ -0,0 +1,208 @@
+      maxGasPrice: parseBigInt(attrs['max-gas-price']),
+      feeFixedUsd: attrs['fee-fixed-usd'],
+      feePercentage: attrs['fee-percentage'],
+      salt: attrs['salt'],
+      spHash: attrs['sp-hash'],
+      chainId: attrs['chain-id'],
+      payAt: parseDate(attrs['pay-at']),
+      recurringDayOfMonth: attrs['recurring-day-of-month'],
+      recurringUntil: parseDate(attrs['recurring-until']),
+      validForDays: attrs['valid-for-days'],
+    } as unknown as ScheduledPayment;
+
+    if (params.recurringDayOfMonth != null) {
+      params.payAt = calculateNextPayAt(new Date(), params.recurringDayOfMonth);
+    }
+
+    let errors = this.scheduledPaymentValidator.validate(params);
+    let hasErrors = Object.values(errors).flatMap((i) => i).length > 0;
+    if (hasErrors) {
+      ctx.status = 422;
+      ctx.body = {
+        errors: serializeErrors(errors),
+      };
+    } else {
+      let prisma = await this.prismaManager.getClient();
+
+      let scheduledPayment = await prisma.scheduledPayment.create({
+        data: params,
+      });
+
+      ctx.body = this.scheduledPaymentSerializer.serialize(scheduledPayment);
+      ctx.status = 201;
+      ctx.type = 'application/vnd.api+json';
+    }
+  }
+
+  async patch(ctx: Koa.Context) {
+    if (!ensureLoggedIn(ctx)) {
+      return;
+    }
+
+    let userAddress = ctx.state.userAddress;
+    let scheduledPaymentId: string = ctx.params.scheduled_payment_id;
+
+    let prisma = await this.prismaManager.getClient();
+    let scheduledPayment = await prisma.scheduledPayment.findFirst({
+      where: {
+        id: scheduledPaymentId,
+        userAddress,
+      },
+    });
+
+    if (scheduledPayment) {
+      let creationTransactionHash = ctx.request.body.data.attributes['creation-transaction-hash'];
+
+      if (creationTransactionHash) {
+        scheduledPayment = await prisma.scheduledPayment.update({
+          where: {
+            id: scheduledPayment.id,
+          },
+          data: {
+            creationTransactionHash,
+          },
+        });
+
+        await this.workerClient.addJob('scheduled-payment-on-chain-creation-waiter', {
+          scheduledPaymentId: scheduledPayment.id,
+        });
+
+        ctx.body = this.scheduledPaymentSerializer.serialize(scheduledPayment);
+        ctx.status = 200;
+      }
+    } else {
+      ctx.status = 404;
+    }
+
+    ctx.type = 'application/vnd.api+json';
+  }
+
+  async delete(ctx: Koa.Context) {
+    if (!ensureLoggedIn(ctx)) {
+      return;
+    }
+
+    let prisma = await this.prismaManager.getClient();
+
+    let userAddress = ctx.state.userAddress;
+    let scheduledPaymentId: string = ctx.params.scheduled_payment_id;
+
+    let scheduledPayment = await prisma.scheduledPayment.findFirstOrThrow({
+      where: {
+        id: scheduledPaymentId,
+        userAddress,
+      },
+    });
+
+    if (scheduledPayment.userAddress !== userAddress) {
```

Comment:
Is this possible, considering that `userAddress` is in the database query?

Reply:
You are correct, good catch! I should replace `findFirstOrThrow` with `findFirst` and use `if (scheduledPayment) {} else {}`